compiler/src/main/scala/org/scalajs/core/compiler/GenJSCode.scala

```diff
@@ -3224,11 +3235,11 @@ abstract class GenJSCode extends plugins.PluginComponent
     }
 
     /** Gen JS code for a call to AnyRef.synchronized */
-    private def genSynchronized(tree: Apply, isStat: Boolean): js.Tree = {
+    private def genSynchronized(tree: Tree, receiver: Tree, arg: Tree,
+        isStat: Boolean): js.Tree = {
```

Comment:
Consider just passing in the position instead of `tree`. Passing in the tree and its destructured form just to get the position seems a bit strange.